src/numeric_index.c

```diff
@@ -160,16 +184,27 @@ double NumericRange_Split(NumericRange *n, NumericRangeNode **lp, NumericRangeNo
 
   RSIndexResult *res = NULL;
   IndexReader *ir = NewMinimalNumericReader(n->entries, false);
+  double split = NumericRange_GetMedian(ir);
+  // if (split == n->minVal || split == n->maxVal) {
+  //   split = n->minVal + (n->maxVal - n->minVal) / 2;
+  // }
+  if (split == n->minVal) {
```

Comment:
Not with the current implementation.
Because of the condition of the splitting, if the median of the min value we will put all the values on the same side (no value is smaller than the median).
This cannot happen with the max value. In this case we will set the max value on one side and the rest on the other